### PR TITLE
Feature/major refactoring

### LINKED ISSUE #8: Test on Safari
Thank you for this component. Have you tested this on Safari? It looks like on Safari, it is not able to read the code from the URL and therefore the popup window is never closed (even though the authentication succeeds).

### LINKED ISSUE #23: Popup doesnt open in Safari on MacOS
I'm using `react-oauth-popup` with NextJS and the popup works fine in Firefox and Chrome but it only shortly appears in Safari and then closes immediately. 

Has anyone of you experienced this issue before?

### PR DESCRIPTION
**Pull request by kgoedecke**
Summary:

- Adds TypeScript
- Adds ESLint
- Remove CircleCI in favor of GitHub Actions
- Use hooks and functional components
- Remove storybook (at least for now)
- Add standard version
- Add commit lint
- Use Yarn instead of npm

Fixes #23 
Fixes #8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,58 +1,59 @@
 {
-  "name": "react-oauth-popup",
-  "version": "0.1.1",
-  "description": "",
+  "name": "react-simple-oauth-popup",
```
Comment (hayyaan) [inline package.json:2]: Are you also going to update the repository name?

File: src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState, useRef } from 'react';
+
+type IWindowProps = {
```
Comment (hayyaan) [inline src/index.tsx:3]: [nitpick] Why not use `interface` for this instead? I guess the I in the name is for Interface anyway :)

File: src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  const onContainerClick = () => {
+    setExternalWindow(createPopup({
+      url, title, width, height,
+    }));
+  };
+
+  useEffect(() => {
+    if (externalWindow) {
+      intervalRef.current = window.setInterval(() => {
```
Comment (hayyaan) [inline src/index.tsx:53]: Are you intending to open a window every 700ms?

File: __tests__/should-close-popup-on-unmount.js [hunk 4]
```diff
@@ -1,27 +1,34 @@
...
-    popup = component.toJSON();
-    expect(global.open.mock.calls.length).toBe(1);
-    component.unmount();
-    expect(close.mock.calls.length).toBe(1);
-    expect(global.open.mock.calls.length).toBe(1);
-});
\ No newline at end of file
+  });
+  popup = component.toJSON();
+  expect(global.open.mock.calls.length).toBe(1);
```
Comment (hayyaan) [inline __tests__/should-close-popup-on-unmount.js:30]: [minor] You can also something like `expect(global.open).toHaveBeenCalled()` See: https://jestjs.io/docs/en/expect#tohavebeencalledtimesnumber

File: __tests__/should-close-popup-on-unmount.js [hunk 5]
```diff
@@ -1,27 +1,34 @@
 import React from 'react';
-import renderer from 'react-test-renderer';
-import OauthPopup from "../src/index.js";
+import renderer, { act } from 'react-test-renderer';
```
Comment (hayyaan) [inline __tests__/should-close-popup-on-unmount.js:2]: [minor] Look into [React Testing Library](https://testing-library.com/docs/react-testing-library/intro), it's quite good for basic usage and it does a lot of the `act` shenanigans for you.

File: package.json (hunk 1 above)
Comment (kgoedecke) [reply package.json:2]: good catch, I already ran into it when releasing it haha.

Comment (kgoedecke) [reply package.json:2]: Fixed in master

File: src/index.tsx (hunk 3 above)
Comment (kgoedecke) [reply src/index.tsx:53]: Every 700ms it checks if the oauth popup was successfully submitted and if a redirect happened. Does that make sense?

**Pull request merged**